modules/javafx.controls/src/main/java/javafx/scene/control/skin/ProgressIndicatorSkin.java

```diff
@@ -506,10 +513,11 @@ public DeterminateIndicator(ProgressIndicator control, ProgressIndicatorSkin s,
             text.setTextOrigin(VPos.TOP);
             text.getStyleClass().setAll("text", "percentage");
 
-            registerChangeListener(text.fontProperty(), o -> {
+            fontListener = o -> {
                 doneTextWidth = Utils.computeTextWidth(text.getFont(), DONE, 0);
                 doneTextHeight = Utils.computeTextHeight(text.getFont(), DONE, 0, TextBoundsType.LOGICAL_VERTICAL_CENTER);
-            });
+            };
+            registerChangeListener(text.fontProperty(), (Consumer<ObservableValue<?>>) fontListener);
 
```

Comment:
wondering why you add a (strong) field reference here? 

Thought that registerChangeListener(observableValue, consumer) takes care of handling storage/cleanup - provided its counter-method unregisterChangeListener(observableValue) is called, which is missing, so seems to be the only part that has to be added. Or not?

Reply:
Yes, that doesn't make a lot of sense. It's probably an artifact because I expected a different API do unregister the listener. I'm gonna change it back!

Reply:
just thinking aloud: actually, it's rather whacky that the indicator (mis-)uses the skin's multiplePropertyListener - it's api isn't meant for frequent register/unregisters (the unregisters is quite a heavy measure). Maybe it should have its own? Or the other way round: skin listening to the property and indicator having a method to update based on font which is invoked by skin's listener?

Reply:
Yes, that would be reasonable.

I would rewrite the fix, add an method "updateFont" to the skin, which also updates the DeterminationIndicator.

When I look into the code, I'm a little bit confused. The fontListeners doesn't get called without a font-change. In this case, the variable doneTextWidth and doneTextHeight never get's initialized. Do I miss something or is the current code wrong?

Reply:
welcome to the jungle :) looks like the code can't really work (and probably is untested) or is not important ..